sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -327,6 +327,13 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val PARQUET_RECORD_FILTER_ENABLED = buildConf("spark.sql.parquet.recordFilter")
+    .doc("Whether to allow the record-level filtering via Parquet API. When " +
+      "'spark.sql.parquet.filterPushdown' is disabled, this configuration does not " +
+      "have any effect.")
```

Comment:
> If true, enable parquet native record level filtering using the pushed down filters. This conf only has an effect when both 'spark.sql.parquet.filterPushdown' and 'spark.sql.parquet.enableVectorizedReader' are enabled.

Reply:
I think we are currently doing only group level filtering if `spark.sql.parquet.enableVectorizedReader` is enabled. Let me write it as "when 'spark.sql.parquet.enableVectorizedReader' is disabled"

Reply:
and even if we enable `spark.sql.parquet.enableVectorizedReader`, we can still fallback if it contains non-`AtomicType`. Let me just take out `' and 'spark.sql.parquet.enableVectorizedReader' are enabled` if you wouldn't mind.